transhumandesign/kag-base
# [added] Chicken/Fishy/Steak turn into food after getting hit by fire arrow + [fixed] fireplace ingredient bug

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

```
[added] turn chicken into food with name "Fried Chicken" by fire arrow hit
[added] turn chicken into food by being in fire for 50 ticks
[added] turn fishy into food by fire arrow hit
[added] turn fishy into food by having it be in fire for every 5th tick 10 times
[added] turn steak into food by fire arrow hit
[fixed] Many fireplaces overlapping each other making many foods from one ingredient
[fixed] Spawned ingredient could be cooked and heal you at the same time
[fixed] Fireplace would not remember that it has been extinguished when rejoining the server
```

Fixes https://github.com/transhumandesign/kag-base/issues/1399
Fixes https://github.com/transhumandesign/kag-base/issues/1909
Fixes https://github.com/transhumandesign/kag-base/issues/1929
Fixes https://github.com/transhumandesign/kag-base/issues/2031

This PR makes it so that chicken turns into a food item with a chicken wing sprite and the name "Fried Chicken" if it is hit by a firearrow or after it has been in fire for 50 ticks.

This gives another use to chicken, is humorous and seems more realistic. I always found it odd that chicken can just get away with being in fire without taking damage. Also adds strategy, if players want to trade one fire arrow for one food.

This PR does not implement all the suggestions from the above linked issue, nor does it add IsFlammable script to chicken. This is just a start to see if this is something worth adding to the game.

~I had trouble compressing Food.png, so if this PR gets accepted someone should compress it more.~
I replaced the png file, now it is 277 bytes smaller compared to the live build (-37%).

There is a bug with many fireplaces (that overlap each other) producing many foods from just one ingredient both in offline and online. This PR fixes that.

Tested offline and online, works as intended.

## Steps to Test or Reproduce

Spawn chicken, shoot it with a fire arrow.
Notice it will turn into a chicken wing.

Build some buildings or wood backtiles and have chicken be closed inside.
Burn it up.
Notice the chicken may turn into a chicken wing if it touches the fire long enough.
